torchgeo/datasets/geo.py

```diff
@@ -286,7 +286,7 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
                     filepath = glob.glob(os.path.join(directory, filename))[0]
                     band_filepaths.append(filepath)
                 data_list.append(self._merge_files(band_filepaths, query))
-            data = torch.stack(data_list)
+            data = torch.cat(data_list)
```

Comment:
Not sure exactly how robust this is, but when `rasterio` reads a single Landsat band, the output dimensions are (1, H, W) instead of (H, W) so we need to concatenate along the first axis instead of stacking along a new axis. For files like NAIP where multiple bands are in the same file, the output dimensions are (B, H, W) as expected.